### PR TITLE
feat(media): extract media from rich message headers

### PR DESCRIPTION
**Pull request by gabrieljablonski**
Template, interactive and buttons messages can carry a media attachment in their header (for example an invoice PDF or a banner image at the top of a template), nested inside the payload instead of at the top level. The bridge missed these, so it served no bytes and the dashboard rendered the rich card without its header media.

This is the bridge half of fazer-ai/chatwoot#315 (WhatsApp rich messages). It is backward compatible: plain media messages are unchanged, and a header with no media still returns null.

## How to test

1. Send a WABA/Cloud-API **template with media in the header** (image or an invoice PDF) to a Baileys-connected number.
2. Confirm the bridge downloads and serves the header attachment (the dashboard, with chatwoot#315, then shows the media inside the rich card).
3. Send a text-only template and confirm nothing regresses (no media served).

## What changed

- `extractMediaMessage` now falls back to a new `extractHeaderMediaMessage`, which unwraps the template (`hydratedFourRowTemplate` / `hydratedTemplate`), interactive (`header`), nested WABA interactive template (`templateMessage.interactiveMessageTemplate.header`) and buttons headers, and surfaces any `imageMessage` / `videoMessage` / `documentMessage`.
- Added spec coverage for each header shape and a text-only no-media case.

- - -
This change is [Reviewable](https://reviewable.io/reviews/fazer-ai/baileys-api/329)

## Summary by CodeRabbit

* **New Features**
  * Improved media handling to detect and download images, videos, and documents embedded in template, interactive, and button message headers; when multiple headers exist, media found in later headers is preferred.

* **Tests**
  * Added unit tests validating header-based media extraction across message types and edge cases, including empty earlier headers and text-only messages (no media returned).

### CODE REVIEW

**Comment by gabrieljablonski** [comment]
@coderabbitai review

**Comment by gabrieljablonski** [comment]
@coderabbitai full review

File: src/baileys/helpers/downloadMediaFromMessages.ts [hunk 1]
```diff
@@ -105,7 +105,46 @@ function extractMediaMessage(message: proto.IMessage): {
...
+function extractHeaderMediaMessage(message: proto.IMessage): {
+  mediaMessage: MediaMessage;
+  mediaType: MediaType;
+} | null {
+  const header =
+    message.templateMessage?.hydratedFourRowTemplate ??
+    message.templateMessage?.hydratedTemplate ??
+    message.interactiveMessage?.header ??
+    message.templateMessage?.interactiveMessageTemplate?.header ??
+    message.buttonsMessage;
```
[the remark this answers is not in the document]
Comment (gabrieljablonski) [reply src/baileys/helpers/downloadMediaFromMessages.ts]: Fixed in 4c823f1 — replaced the `??` short-circuit with a loop that scans every candidate header container (`hydratedFourRowTemplate`, `hydratedTemplate`, `interactiveMessage.header`, `interactiveMessageTemplate.header`, `buttonsMessage`) and returns the first one that actually carries media, so a media-less earlier container no longer masks media in a later one. Added a precedence regression test ("prefers later header media when earlier header has no media").

**Comment by gabrieljablonski** [comment]
@coderabbitai full review

**Comment by gabrieljablonski** [comment]
@coderabbitai full review

**Comment by gabrieljablonski** [comment]
@coderabbitai full review

**Comment by gabrieljablonski** [comment]
@coderabbitai full review
